src/main/java/org/verdictdb/coordinator/QueryResultAccuracyEstimatorFromDifference.java

```diff
@@ -0,0 +1,122 @@
+package org.verdictdb.coordinator;
+
+import org.verdictdb.VerdictResultStream;
+import org.verdictdb.VerdictSingleResult;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+public class QueryResultAccuracyEstimatorFromDifference extends QueryResultAccuracyEstimator {
+
+  Coordinator runningCoordinator;
+
+  // the values of the result should be within [(1-valueError)*prevValue, (1+valueError)*prevValue] of the previous result.
+  // Otherwise, it will fetch next result.
+  Double valueError = 0.05;
+
+  // the #row of the result should be within  [(1-groupCountError)*prev#row, (1+groupCountError)*prev#row] of the previous result.
+  // Otherwise, it will fetch next result.
+  Double groupCountError = 0.05;
+
+  // key is the values of non-aggregated column, value is the values of aggregated column
+  HashMap<List<Object>, List<Object>> aggregatedMap = new HashMap<>();
+
+  QueryResultAccuracyEstimatorFromDifference(Coordinator runningCoordinator) {
+    this.runningCoordinator = runningCoordinator;
+  }
+
+  public void setValueError(Double valueError) {
+    this.valueError = valueError;
+  }
+
+  public void setGroupCountError(Double groupCountError) {
+    this.groupCountError = groupCountError;
+  }
+
+  /**
+   * fetch the answer from stream until converge
+   *
+   * @return the accurate answer
+   */
+  @Override
+  public boolean isLastResultAccurate() {
+    if (!checkConverge()) {
+      return false;
+    } else {
+      log.debug("Break condition has reached.");
+      log.debug("Aborts an ExecutionContext: " + this);
+      if (runningCoordinator != null) {
+        Coordinator c = runningCoordinator;
+        runningCoordinator = null;
+        c.abort();
```

Comment:
I found it here. Can we move this to the upper level, i.e., ExecutionContext? Since the coordinator was created in ExecutionContext, it is natural that ExecutionContext closes or aborts it.

Reply:
Got it. I will move it to ExecutionContext.